doc/build/kconfig/setting.rst

```diff
@@ -149,7 +149,7 @@ used.
 
    3. From the CMake variable cache
 
-2. Otherwise if ``CONF_FILE`` is set, and a single configuration file of the
+2. Otherwise if ``CONF_FILE`` is not set, and a single configuration file of the
```

Comment:
This is not an error; "if `CONF_FILE` is set" is correct.

But I understand how you might've thought it was an error, and I think this section could be reworded a bit to avoid confusing readers. So, thank you for making the observation.

What point 2 is meant to convey is that if you pass `-DCONF_FILE=prj_<build>.conf`, the build system will recognize that exact form and use the `<build>` substring to find additional files. Point 2 is really a special case of point 1.

There happens to be one example of this working in the Zephyr tree. If you build:
```
west build -b nucleo_h743zi samples/subsys/fs/littlefs -DCONF_FILE=prj_blk.conf
```
then both `prj_blk.conf` and an extra, board-specific file with the `blk` suffix will be picked up:
```
Merged configuration '(...)/zephyr/samples/subsys/fs/littlefs/prj_blk.conf'
Merged configuration '(...)/zephyr/samples/subsys/fs/littlefs/boards/nucleo_h743zi_blk.conf'
```